test/services/katello/pulp3/repository/python/python_test.rb

```diff
@@ -0,0 +1,47 @@
+require 'katello_test_helper'
+require 'support/pulp3_support'
+
+module Katello
+  module Service
+    module Pulp3
+      class Repository
+        class PythonTest < ActiveSupport::TestCase
+          include Katello::Pulp3Support
+          include RepositorySupport
+
+          def setup
+            @repo = katello_repositories(:pulp3_python_1)
+            @primary = FactoryBot.create(:smart_proxy, :default_smart_proxy, :with_pulp3)
+
+            create_repo(@repo, @primary)
+
+            @repo.reload
+          end
+
+          def test_update
+            ForemanTasks.sync_task(
+              ::Actions::Pulp3::Repository::UpdateRepository, @repo, @primary)
+          end
+
+          def test_delete
+            repo_reference = Katello::Pulp3::RepositoryReference.find_by(
+              root_repository_id: @repo.root.id,
+              content_view_id: @repo.content_view.id)
+
+            refute_nil repo_reference
+
+            ForemanTasks.sync_task(
+              ::Actions::Pulp3::Repository::Delete, @repo.id, @primary)
+            @repo.reload
+
+            repo_reference = Katello::Pulp3::RepositoryReference.find_by(
+              root_repository_id: @repo.root.id,
+              content_view_id: @repo.content_view.id)
+
+            assert_nil repo_reference
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
Something i've kinda thought about is to create a test framework to make fully testing a generic repo type (or maybe even other types) quite a bit simpler.  

I imagine it would be something like:

```
Class MytestClass   
   include GenericRepoTesting
    
    test_generic_type  'python'
    test_sync 'https://someserverurl/', :other_repository_option => 'somevalue'
    test_upload './path/to/sample/file/to/upload'
end
```

lets not do it in this Pr, but maybe something to add to work on in the future :)

Reply:
also keep in mind 'include GenericRepoTesting' here has all the magic